### PR TITLE
[dbnode] Increase readerDecoderStream.buf length to 64

### PR DESCRIPTION
**Pull request by linasm**
**What this PR does / why we need it**:
I've noticed that `readerDecoderStream` is using [a very small (8 bytes) buffer](https://github.com/m3db/m3/blob/18430610ed4501dcef1b8da39291dd6cc527681c/src/dbnode/persist/fs/read_decode_stream.go#L44) which does not perform very well with `adler32` hash implementation. Increasing this buffer to 64 bytes gives a very noticeable performance boost for `DataFileSetReader`.

**Special notes for your reviewer**:
I've chosen 64 bytes as a compromise between buffer size and diminishing performance improvements.
Benchmarking results using `read_data_files` utility in `series` benchmarking mode (no datapoint decoding):
```
read_data_files -b 1596009600000000000 -p ~/testdata/m3db -s 4 -B series
```
| len(buf) | running time |
|----|----|
| 8 | 42 ms |
| 16 | 36 ms |
| 32 | 31 ms |
| 64 | 28 ms |
| 128 | 27 ms |
| 256 | 26 ms |

**Does this PR introduce a user-facing and/or backwards incompatible change?**:
NONE

**Does this PR require updating code package or user-facing documentation?**:
NONE

### CODE REVIEW

File: src/dbnode/persist/fs/read_decode_stream.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ type readerDecoderStream struct {
 	bytesReader      *bytes.Reader
 	readerWithDigest digest.ReaderWithDigest
 	backingBytes     []byte
-	buf              [8]byte
+	buf              [64]byte
```
Comment (vdarulis) [inline src/dbnode/persist/fs/read_decode_stream.go:44]: it looks like your result matches expected cache line size, maybe worth setting this to `len(cpu.CacheLinePad)` from x/sys?

**Review by vdarulis — APPROVED** [review]
lgtm with comment

File: src/dbnode/persist/fs/read_decode_stream.go (hunk 1 above)
Comment (linasm) [reply src/dbnode/persist/fs/read_decode_stream.go:44]: Thanks for pointing me to this interesting feature, was not aware of it.
However, I see that this value can potentially be 0 (see `const cacheLineSize = 0` in cpu_wasm.go) which would elevate the issue from performance to correctness one. Also, all the practical target architectures (intel + arm64) have it at 64, and I have no way to benchmark the effect on anything else, so I guess I'll just keep it simple and stick to a fixed number.

Comment (vdarulis) [reply src/dbnode/persist/fs/read_decode_stream.go:44]: gotcha!

**Pull request merged**